src/mame/apple/apple2.cpp

```diff
@@ -481,7 +481,7 @@ u8 apple2_state::flags_r(offs_t offset)
 		// check if SHIFT key mod configured
 		if (m_sysconfig->read() & 0x04)
 		{
-			return ((m_gameio->sw2_r() || (m_kbspecial->read() & 0x06)) ? 0x80 : 0) | uFloatingBus7;
+			return ((m_gameio->sw2_r() || (m_kbspecial->read() & 0x06)) ? 0 : 0x80) | uFloatingBus7;
```

Comment:
It seems that this will also invert the effect of the actual button 2 (`m_gameio->sw2_r()`). I think a more appropriate fix would be

```suggestion
			return ((m_gameio->sw2_r() || !(m_kbspecial->read() & 0x06)) ? 0x80 : 0) | uFloatingBus7;
```

Reply:
I guess if you were holding down the shift key and then pressed pb2 it might go back to high, so the existing logic is fine.

Reply:
@arekkusu42 Okay, makes sense. Does this mean that the values on line 486 (the non-Shift-key-mod branch) should be inverted as well?

Reply:
apple3 doesn't have a cassette.  apple1's cassette works somewhat differently and doesn't have issues reading tapes as far as I'm aware.